mockintosh/stats.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+"""
+.. module:: __init__
+    :synopsis: module that contains server classes.
+"""
+
+
+class BaseStats:
+    def __init__(self):
+        self.parent = None
+        self.children = None
+        self.request_counter = 0
+        self.status_code_distribution = {}
+        self.avg_response_time_in_microseconds = 0
+
+    def increase_request_counter(self):
+        self.request_counter += 1
+        if not hasattr(self, 'parent'):
+            return
+
+        method = getattr(self.parent, "increase_request_counter", None)
+        if callable(method):
+            method()
+
+    def add_request_elapsed_time(self, elapsed_time_in_microseconds):
+        self.avg_response_time_in_microseconds = (
+            elapsed_time_in_microseconds + (
+                self.avg_response_time_in_microseconds * self.request_counter
+            )
+        ) / (self.request_counter + 1)
+        if not hasattr(self, 'parent'):
+            return
+
+        method = getattr(self.parent, "update_elapsed_time", None)
+        if callable(method):
+            method()
+
+    def update_elapsed_time(self):
+        if self.request_counter == 0:
+            return
+
+        total_elapsed_time = 0
+        if hasattr(self, 'services'):
+            for child in self.services:
+                total_elapsed_time += child.avg_response_time_in_microseconds * child.request_counter
+        if hasattr(self, 'endpoints'):
+            for child in self.endpoints:
+                total_elapsed_time += child.avg_response_time_in_microseconds * child.request_counter
+        self.avg_response_time_in_microseconds = total_elapsed_time / self.request_counter
+
+        if not hasattr(self, 'parent'):
+            return
+
+        method = getattr(self.parent, "update_elapsed_time", None)
+        if callable(method):
+            method()
+
+    def add_status_code(self, status_code):
+        if status_code not in self.status_code_distribution:
+            self.status_code_distribution[status_code] = 1
+        else:
+            self.status_code_distribution[status_code] += 1
+        if not hasattr(self, 'parent'):
+            return
+
+        method = getattr(self.parent, "add_status_code", None)
+        if callable(method):
+            method(status_code)
+
+    def json(self):
+        data = {}
+
+        if hasattr(self, 'identifier'):
+            data['identifier'] = self.identifier
```

Comment:
Let's rename `identifier` in JSON into `comment` or `hint`, because it does not identify the item, just hits what it is

Reply:
Fixed with https://github.com/up9inc/mockintosh/commit/4a37b5c04a11b0f08519cf61be5ef3f340541dca